Document security reviewer conflict of interest

lumjjb opened the pull request:
To address https://github.com/cncf/sig-security/issues/156

Signed-off-by: Brandon Lum <lumjjb@gmail.com>

JustinCappos commented on assessments/guide/security-reviewer.md:
```diff
@@ -43,4 +43,31 @@ expected to conduct an in-depth review described in the [Security Assessment
+Soft conflicts:
+- Reviewer belongs to the same company/organization of the project, but does not work on the project
+- Reviewer uses the project in his/her work
+- Reviewer has a personal stake in the project (investments, personal relationships, etc.)
```
Investments seem pretty serious to me.  I can understand personal friendship with maintainers, but financial benefit seems like a bigger motivation for abuse.

tamalsaha replied:
I 2nd this. If the reviewer has financial interest in a project that is a hard conflict.

lumjjb replied:
Yea i think i agree with the financial interest, but having some trouble coming up with proper phrasing for it...

I'm thinking about the case where someone owns a small amount of stock in a large company or an index. i.e. someone invests in S&P index, which encompass most of the bigger tech firms... I think investment in enterpriseDB vs a review of PostgresDB would probably be a hard conflict but investing in S&P shouldn't be a gatekeeper to someone reviewing keycloak.

ultrasaurus replied:
Financial interest is strange because this doesn't apply directly to open source projects, and at its extreme, I wouldn't have thought that having stock in Google represents conflict of with review of Kubernetes.

ultrasaurus replied:
I think the change @lumjjb added addresses this is nicely in simple language

JustinCappos commented on assessments/guide/security-reviewer.md:
```diff
@@ -43,4 +43,31 @@ expected to conduct an in-depth review described in the [Security Assessment
+- Reviewer is paid to work on the project
+
+Soft conflicts:
+- Reviewer belongs to the same company/organization of the project, but does not work on the project
```
This is a hard conflict in academic terms.  (I'm just saying, not making a recommendation for change.)

lumjjb replied:
I think this is a good topic for discussion. I think it depends quite a bit on how much we think the project lead has influence over the outcome of the security review as well. 

I'm also wondering now if we need to define some rules of projects that are under open governance as well.

ultrasaurus replied:
Would be interested in hearing more about academic conflict.  

I was also thinking this would be a "soft" conflict -- given there are multiple reviewers and someone from security team of a company who is involved with the project would have great context to contribute.

ultrasaurus commented on assessments/guide/security-reviewer.md:
```diff
@@ -43,4 +43,31 @@ expected to conduct an in-depth review described in the [Security Assessment
+guidelines for conflict of interest situations are important to prevent:
+
+- Individuals from intentionally or unintentionally promote their own company's project
+- Sig-security chairs and assessment leads could intentionally or unintentionally limit the participation of an individual unfairly by asserting conflict of interest
```
nit: SIG-Security

ultrasaurus commented on assessments/guide/security-reviewer.md:
```diff
@@ -43,4 +43,31 @@ expected to conduct an in-depth review described in the [Security Assessment
+
+Soft conflicts:
+- Reviewer belongs to the same company/organization of the project, but does not work on the project
+- Reviewer uses the project in his/her work
```
- Reviewer has contributed to the project.

both this and using the project could be a strong benefit.

lumjjb commented:
From meeting 08/07: Create a process to evaluate conflicts by chairs

pragashj commented on assessments/guide/security-reviewer.md:
```diff
@@ -43,4 +43,35 @@ expected to conduct an in-depth review described in the [Security Assessment
+
+- Individuals from intentionally or unintentionally promote their own company's project
+- SIG-Security chairs and assessment leads could intentionally or unintentionally limit the participation of an individual unfairly by asserting conflict of interest
+- Security reviews being stalled while groups discuss whether this person or that person should be allowed to participate
```
Just a nit
Instead of *discuss whether this person or that person* how about "groups belabor on who"

pragashj commented on assessments/guide/security-reviewer.md:
```diff
@@ -43,4 +43,35 @@ expected to conduct an in-depth review described in the [Security Assessment
+security assessment documentation. It is the responsibility of the security reviewer
+to make known his/her own conflict of interests.
+
+For each project, 2 SIG-Security chairs must sign off on the conflicts presented to them that the assessment lead has no conflicts, and reviewers have no hard conflicts.
```
As much as possible we try to avoid conflict, but in cases where we need to go with a conflict, we should document the justification for allowing the conflict. This could be as specific as "This reviewer is the expert in this field and therefore we require their input" or as conventional as "No one else has the bandwidth to take this on at this time so we are letting an exception" -- over time, if we are documenting and reviewing the reason for exception it will allow us to keep the process honest.

lumjjb commented:
@pragashj  addressed comments!

>@lumjjb lumjjb dismissed stale reviews from pragashj and ultrasaurus via cb086ba 9 minutes ago

It seems like a setting is enabled to dismiss LGTMs on code pushes? @ultrasaurus @pragashj could you review it again? 

Do we want to keep this setting? Seems like it's not critical for us to ensure that reviews need to be closely up to date.

ultrasaurus commented:
@lumjjb agreed on the setting... changed to not required.  Will take a look at the changes, since it's not retractive!
